clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6350,6 +6350,77 @@ static void handleSYCLIntelPipeIOAttr(Sema &S, Decl *D,
   S.addSYCLIntelPipeIOAttr(D, Attr, E);
 }
 
+SYCLIntelFPGAMaxConcurrencyAttr *Sema::MergeSYCLIntelFPGAMaxConcurrencyAttr(
+    Decl *D, const SYCLIntelFPGAMaxConcurrencyAttr &A) {
+  // Check to see if there's a duplicate attribute with different values
+  // already applied to the declaration.
+  if (const auto *DeclAttr = D->getAttr<SYCLIntelFPGAMaxConcurrencyAttr>()) {
+    const auto *DeclExpr = dyn_cast<ConstantExpr>(DeclAttr->getNThreadsExpr());
+    const auto *MergeExpr = dyn_cast<ConstantExpr>(A.getNThreadsExpr());
+    if (DeclExpr && MergeExpr &&
+        DeclExpr->getResultAsAPSInt() != MergeExpr->getResultAsAPSInt()) {
+      Diag(DeclAttr->getLocation(),
+           diag::err_sycl_loop_or_function_attr_duplication)
```

Comment:
@AaronBallman , I am wondering whether we can use diag::warn_duplicate_attribute here since we are using this inside Add function.

Reply:
Good catch! Warning would be more consistent with the other attributes.

Reply:
@smanna12  and @AaronBallman  thanks for the review. So should I be removing the err_sycl_loop_or_function_attr_duplication and revert to the old name err_sycl_loop_attr_duplication since it isn't needed now? Or leave it as is?

Reply:
In the other cases, the way we handle duplicate attributes is to see if it's an exact duplication where the args have identical values and silently ignore the duplicate in that case, or warn the user about the duplication and drop the second attribute (retaining the first one).

So I guess my question really boils down to: is there a need for this issue to stop the user's compilation? My personal view is that there's nothing special about this particular attribute and so there's no real need for an error. Based on that, I'd recommend changing to use `diag::warn_duplicate_attribute` and note the duplication, as with the other SYCL attributes.

Reply:
Agree. 
My question was about the code in SemaStmtAttr.cpp at line 630. Should it be reverted to:

 for (const auto *I : Attrs) {
    if (LoopAttr && isa<LoopAttrT>(I)) {
      // Cannot specify same type of attribute twice.
      S.Diag(I->getLocation(), diag::**err_sycl_loop_or_function_attr_duplication**)
          << isIntelFPGAAttr << 0 << LoopAttr;
    }
    if (isa<LoopAttrT>(I))
      LoopAttr = cast<LoopAttrT>(I);
  }
since I will not be using **err_sycl_or_function_attr_duplication**? Or should I leave the name as I changed it in this PR?

Reply:
Ohhh, I see!

For the *statement* attribute, I think an error is appropriate because the user had to work really hard to do something that baffling. I'd roll back the changes to DiagnosticSemaKinds.td and continue to use `err_sycl_loop_attr_duplication` for the duplicate statement attribute, and use `diag::warn_duplicate_attribute` for the declaration attribute.

WDYT?

Reply:
>>I'd roll back the changes to DiagnosticSemaKinds.td and continue to use err_sycl_loop_attr_duplication for the duplicate statement attribute, and use diag::warn_duplicate_attribute for the declaration attribute.

Yes, i agree.